### PR TITLE
Fallback to ENV vars if args are not set

### PR DESCRIPTION
**Pull request by unbelauscht**
Fallback to ENV vars if args are not set.

### CODE REVIEW

**Comment by dan-farmer** [comment]
Hi @unbelauscht. Thanks for the contribution!

I had a quick look and I'm not sure the problem this solves - could you elaborate a bit?

Currently, if `--profile` and `--region` args are omitted, `boto_session` will be created as `boto3.Session(profile_name=None, region_name=None)`. This matches the default behaviour if `profile_name`/`region_name` are not provided (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session), and will then fall through to the default configuration logic path. This includes the standard environment variables (https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-environment-variables), such as `AWS_PROFILE` and `AWS_DEFAULT_REGION`.

Some confusion exists on the latter, because some AWS SDKs recognise `AWS_DEFAULT_REGION`, some recognise `AWS_REGION`, and some both. `AWS_DEFAULT_REGION` is the documented config for boto3 specifically (see last link).

So in summary, I think this already works - provided you use the boto3 documented `AWS_DEFAULT_REGION` env var, not `AWS_REGION`.

(Disclaimer: It's been a while since I looked at this; I could have misremembered how it behaves and/or be misinterpreting the docs.)

**Comment by unbelauscht** [comment]
Hi @dan-farmer,

I wasn't aware AWS_DEFAULT_REGION is a thing.

Terraform e.g. also recognises AWS_REGION.

Feel free to close my pull request.

**Comment by dan-farmer** [comment]
In fairness, as I mentioned the use of `AWS_DEFAULT_REGION` vs `AWS_REGION` is inconsistent between the different AWS language SDKs. Presumably Terraform uses the AWS Go SDK, and inherits `AWS_REGION` from there: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-the-region
